client-core/src/cipher/default.rs

```diff
@@ -143,14 +141,11 @@ impl TransactionObfuscation for DefaultTransactionObfuscation {
                     ))
                 }
             };
-            let request = SECP.with(|secp| {
-                DecryptionRequest::create(
-                    &secp,
-                    transaction_ids.to_owned(),
-                    ch,
-                    &private_key.into(),
-                )
-            });
+            let pk = match private_key_kind {
+                PrivateKeyKind::Normal(p) => p.into(),
+            };
```

Comment:
What happens for the hardware kind here, which I think not able to return private key?

Reply:
this is for the view key -- I guess if this is implementing the "mixed" mode (where this will always be local), then it doesn't make sense to change `private_key: &PrivateKey` to that wrapper type?